mpt/src/branch_rlc_init.rs

```diff
@@ -0,0 +1,173 @@
+use halo2_proofs::{
+    plonk::{Advice, Column, ConstraintSystem, Expression, VirtualCells, Fixed},
+    poly::Rotation,
+};
+use eth_types::Field;
+use std::marker::PhantomData;
+use crate::{
+    helpers::range_lookups,
+    mpt::{FixedTableTag, MainCols},
+    param::HASH_WIDTH,
+};
+
+
+#[derive(Clone, Debug)]
+pub(crate) struct BranchRLCInitConfig<F> {
+    _marker: PhantomData<F>,
+}
+
+// BranchRLCInitChip verifies the random linear combination for the branch init
+// row. The rest of random linear combination is checked in branch_acc, the
+// whole RLC is used to check the hash of a branch.
+impl<F: Field> BranchRLCInitConfig<F> {
+    pub fn configure(
+        meta: &mut ConstraintSystem<F>,
+        q_enable: impl Fn(&mut VirtualCells<'_, F>) -> Expression<F> + Copy,
+        s_main: MainCols,
+        acc_s: Column<Advice>,
+        acc_mult_s: Column<Advice>,
+        acc_c: Column<Advice>,
+        acc_mult_c: Column<Advice>,
+        acc_r: F,
+        fixed_table: [Column<Fixed>; 3],
+    ) -> BranchRLCInitConfig<F> {
+        let config = BranchRLCInitConfig { _marker: PhantomData, }; 
+
+        // Short RLP, meta data contains two bytes: 248, 81
+        // [1,0,1,0,248,81,0,248,81,0,3,0,0,0,...
+        // The length of RLP stream is: 81.
```

Comment:
All such information is stored in the `S` part in the branch init row (also for example whether the branch `S` or branch `C` are placeholders ...). Agree, it might be better to store `C` related info into `C` part, but then there is also information like `drifted_pos` which applies to both parts, so there would be some asymmetry in storing these fields in any case. But might be changed later, it should be easy.